pkg/ingester/retention.go

```diff
@@ -290,6 +290,9 @@ func (dc *diskCleaner) CleanupBlocksWhenHighDiskUtilization(ctx context.Context)
 
 // isBlockDeletable returns true if this block can be deleted.
 func (dc *diskCleaner) isBlockDeletable(block *tenantBlock) bool {
+	// TODO(kolesnikovae):
+	//  Expiry defaults to -querier.query-store-after which should be deprecated,
+	//  blocks-storage.bucket-store.ignore-blocks-within can be used instead.
 	expiryTs := time.Now().Add(-dc.policy.Expiry)
 	return block.Uploaded && ulid.Time(block.ID.Time()).Before(expiryTs)
```

Comment:
However, we also need to take into account that a new block does not become available to the store gateway immediately. I'd say that it probably makes sense to delay till expiration of `blocks-storage.bucket-store.ignore-blocks-within` + 2*(block sync interval)